### ISSUE TITLE
[brotli] Build error

### ISSUE THREAD

**Issue by 21020757** [microsoft/vcpkg#24276]
**Host Environment**
- OS: windows 
- Compiler: revision

**To Reproduce**
`./vcpkg install sdl2-ttf:x64-windows`

**Failure logs**
Computing installation plan...
The following packages will be built and installed:
  * brotli[core]:x64-windows -> 1.0.9#2
  * bzip2[core]:x64-windows -> 1.0.8#2
  * freetype[brotli,bzip2,core,png,zlib]:x64-windows -> 2.11.1
    sdl2-ttf[core]:x64-windows -> 2.0.15#5
Additional packages (*) will be modified to complete this operation.
Detecting compiler hash for triplet x64-windows...
Restored 0 packages from C:\Users\fpt\AppData\Local\vcpkg\archives in 417.9 us. Use --debug to see more details.
Starting package 1/4: brotli:x64-windows
Building package brotli[core]:x64-windows...
-- Using cached google-brotli-e61745a6b7add50d380cfd7d3883dd6c62fc2c71.tar.gz.
-- Cleaning sources at C:/Users/fpt/vcpkg/buildtrees/brotli/src/6c62fc2c71-7c6c2710e9.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source C:/Users/fpt/vcpkg/downloads/google-brotli-e61745a6b7add50d380cfd7d3883dd6c62fc2c71.tar.gz
-- Applying patch install.patch
-- Applying patch fix-arm-uwp.patch
-- Applying patch pkgconfig.patch
-- Applying patch fix-ios.patch
-- Using source at C:/Users/fpt/vcpkg/buildtrees/brotli/src/6c62fc2c71-7c6c2710e9.clean
-- Configuring x64-windows
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:128 (message):
    Command failed: ninja -v
    Working Directory: C:/Users/fpt/vcpkg/buildtrees/brotli/x64-windows-rel/vcpkg-parallel-configure
    Error code: 1
    See logs for more information:
      C:\Users\fpt\vcpkg\buildtrees\brotli\config-x64-windows-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_cmake.cmake:345 (vcpkg_execute_required_process)
  ports/brotli/portfile.cmake:14 (vcpkg_configure_cmake)
  scripts/ports.cmake:145 (include)

Error: Building package brotli:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `git pull` and `.\vcpkg update`.
Then check for known issues at:
  https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+brotli
You can submit a new issue at:
  https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[brotli]+Build+error
including:
  package: brotli[core]:x64-windows -> 1.0.9#2
    vcpkg-tool version: 2022-03-09-1affd32f93b299d5a907816c328ca3ededb73a7e
    vcpkg-scripts version: 6b6f5a82d 2022-03-16 (5 weeks ago)

Additionally, attach any relevant sections from the log files above.

**Additional context**
here is the log file
[config-x64-windows-out.log](https://github.com/microsoft/vcpkg/files/8523710/config-x64-windows-out.log)

pls help, thank u

**Comment by FrankXie05** [comment]
````
CMake Error at C:/Users/fpt/vcpkg/downloads/tools/cmake-3.22.2-windows/cmake-3.22.2-windows-i386/share/cmake-3.22/Modules/CMakeTestCCompiler.cmake:69 (message):
  The C compiler

    "C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.31.31103/bin/Hostx64/x64/cl.exe"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: C:/Users/fpt/vcpkg/buildtrees/brotli/x64-windows-dbg/CMakeFiles/CMakeTmp
    
    Run Build Command(s):C:/Users/fpt/vcpkg/downloads/tools/ninja/1.10.2-windows/ninja.exe cmTC_67505 && [1/2] Building C object CMakeFiles\cmTC_67505.dir\testCCompiler.c.obj
    [2/2] Linking C executable cmTC_67505.exe
    FAILED: cmTC_67505.exe 
    cmd.exe /C "cd . && C:\Users\fpt\vcpkg\downloads\tools\cmake-3.22.2-windows\cmake-3.22.2-windows-i386\bin\cmake.exe -E vs_link_exe --intdir=CMakeFiles\cmTC_67505.dir --rc=rc --mt=CMAKE_MT-NOTFOUND --manifests  -- C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\Tools\MSVC\1431~1.311\bin\Hostx64\x64\link.exe /nologo CMakeFiles\cmTC_67505.dir\testCCompiler.c.obj  /out:cmTC_67505.exe /implib:cmTC_67505.lib /pdb:cmTC_67505.pdb /version:0.0 /machine:x64  /debug /INCREMENTAL /subsystem:console  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib && cd ."
    RC Pass 1: command "rc /fo CMakeFiles\cmTC_67505.dir/manifest.res CMakeFiles\cmTC_67505.dir/manifest.rc" failed (exit code 0) with the following output:
    The system cannot find the file specified
    ninja: build stopped: subcommand failed.
    
    

  

  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:7 (project)


-- Configuring incomplete, errors occurred!
See also "C:/Users/fpt/vcpkg/buildtrees/brotli/x64-windows-dbg/CMakeFiles/CMakeOutput.log".
See also "C:/Users/fpt/vcpkg/buildtrees/brotli/x64-windows-dbg/CMakeFiles/CMakeError.log".
ninja: build stopped: subcommand failed.
````

**Comment by FrankXie05** [comment]
@21020757 Thanks for posting this issue, see the required Visual Studio components in https://github.com/microsoft/vcpkg-tool/pull/314.

**Comment by 21020757** [comment]
I did it but now I got another problem

CMake Error at scripts/cmake/vcpkg_copy_tool_dependencies.cmake:52 (message):
  Could not find PowerShell Core; please open an issue to report this.
Call Stack (most recent call first):
  ports/brotli/portfile.cmake:24 (vcpkg_copy_tool_dependencies)
  scripts/ports.cmake:145 (include)

**Comment by FrankXie05** [comment]
@21020757 This problem should be fixed in powershell 7.1.0 version. 
 Related issue: #15681

**Comment by FrankXie05** [comment]
I think this issue hasn’t been fexed; if it is still an issue, please reopen this issue.
